doc/src/manual/code-loading.md

```diff
@@ -0,0 +1,324 @@
+
+1. `/home/me/.julia/packages/Priv/HDkr/src/Priv.jl`
+2. `/usr/local/julia/packages/Priv/HDkr/src/Priv.jl`
+
+Julia uses the first of these that exists to load the public `Priv` package.
+
+Here is a materialized `paths` map for the `App` project environment:
+
+```julia
+paths = Dict{Tuple{UUID,Symbol},String}(
+    # Priv – the private one:
+    (UUID("ba13f791-ae1d-465a-978b-69c3ad90f72b"), :Priv) =>
+        # relative entry-point inside `App` repo:
+        "/home/me/projects/App/deps/Priv/src/Priv.jl",
+    # Priv – the public one:
+    (UUID("2d15fe94-a1f7-436c-a4d8-07a9a496e01c"), :Priv) =>
+        # package installed in the user depot:
+        "/usr/local/julia/packages/Priv/HDkr/src/Priv.jl",
+    # Pub:
+    (UUID("ba13f791-ae1d-465a-978b-69c3ad90f72b"), :Pub) =>
+        # package installed in the system depot:
+        "/home/me/.julia/packages/Pub/oKpw/src/Pub.jl",
+    # Zebra:
+    (UUID("f7a24cb4-21fc-4002-ac70-f0e3a0dd3f62"), :Zebra) =>
+        # package installed in the user depot:
+        "/usr/local/julia/packages/Zebra/me9k/src/Zebra.jl",
+)
+```
+
+This example map includes three different kinds of package locations:
+
+1. The private `Priv` package is "[vendored](https://stackoverflow.com/a/35109534/659248)" inside inside of `App` repository.
+2. The public `Priv` and `Zebra` packages are in the system depot, where packages installed and managed by the system administrator live. These are available to all users on the system.
+3. The `Pub` package is in the user depot, where packages installed by the user live. These are only available to the user who installed them.
+
+### Package directories
+
+Package directories provide a kind of environment that approximates package loading in Julia 0.6 and earlier, and which resembles package loading in many other dynamic languages. The set of packages available in a package directory corresponds to the set of subdirectories it contains that look like packages: if `X/src/X.jl` is a file in a package directory, then `X` is considered to be a package and `X/src/X.jl` is the file you load to get `X`. Which packages can "see" each other as dependencies depends on whether they contain project files or not and what appears in the `[deps]` sections of those project files.
+
+**The roots map** is determined by the subdirectories of a package directory for which `X/src/X.jl` exists and whether `X/Project.toml` exists and has a top-level `uuid` entry. Specifically `:x => uuid` goes in `roots` for each such `X` where `uuid` is defined as:
```

Comment:
I think the last `X` here should be lowercase, to refer to the placeholder in `:x => uuid`? Was a bit confusing along with package name `X`.

Reply:
The lowercase `:x` is just a typo — it should be `:X`.